packages/nuxt/src/pages/module.ts

```diff
@@ -185,56 +185,44 @@ export default defineNuxtModule({
     }
 
     if (useExperimentalTypedPages) {
-      const declarationFile = 'types/typed-router.d.ts'
+      const declarationFile = './types/typed-router.d.ts'
 
       const typedRouterOptions: TypedRouterOptions = {
         routesFolder: [],
         dts: resolve(nuxt.options.buildDir, declarationFile),
         logs: nuxt.options.debug && nuxt.options.debug.router,
         async beforeWriteFiles (rootPage) {
           rootPage.children.forEach(child => child.delete())
-
           const pages = nuxt.apps.default?.pages || await resolvePagesRoutes(options.pattern, nuxt)
-          const augmentPagesContext = nuxt._augmentPagesContext ?? {}
-
           if (nuxt.apps.default) {
             nuxt.apps.default.pages = pages
           }
-
           const addedPagePaths = new Set<string>()
-
           function addPage (parent: EditableTreeNode, page: NuxtPage) {
-            const pathBeforeAugmented = page.file
-              ? augmentPagesContext.augmentedPages?.get(page.file)?.originalPath
-              : undefined
-
-            const path = pathBeforeAugmented ?? page.path
-
             // Avoid duplicate keys in the generated RouteNamedMap type
-            const absolutePagePath = hasLeadingSlash(path)
-              ? path
-              : joinURL(parent.path, path)
+            const absolutePagePath = joinURL(parent.path, page.path)
 
+            // way to add a route without a file, which must be possible
             const route = addedPagePaths.has(absolutePagePath)
               ? parent
-              // @ts-expect-error TODO: either fix types upstream or figure out another
-              // way to add a route without a file, which must be possible
-              : parent.insert(path, page.file)
+              : /^\//.test(page.path)
+                // @ts-expect-error TODO: either fix types upstream or figure out another
+                // way to add a route without a file, which must be possible
+                ? rootPage.insert(page.path, page.file)
+                // @ts-expect-error TODO: either fix types upstream or figure out another
+                // way to add a route without a file, which must be possible
+                : parent.insert(page.path, page.file)
 
             addedPagePaths.add(absolutePagePath)
-
             if (page.meta) {
               route.addToMeta(page.meta)
             }
-            if (page.alias !== undefined) {
+            if (page.alias) {
```

Comment:
Possible edge case: what if you don't have an `index.vue`, but some other page has `alias` set to `''`?

Reply:
that seems like a different PR, perhaps? Shouldn't alias be set to `/` rather than an empty string?

Reply:
Oh that's right 👍🏼